### ISSUE TITLE
Support for recursive inner/outer

### ISSUE THREAD

**Issue by adrpo** [OpenModelica/OpenModelica#10918]
### Description

It seems this pattern is used in TILMedia, used by TransiEnt:
https://libraries.openmodelica.org/branches/master/TransiEnt/TransiEnt.html
See for example:
`[/home/hudson/saved_omc/libraries/.openmodelica/libraries/TILMedia 1.8.1-main/VLEFluidTypes/BaseVLEFluid.mo:4:3-6:42:writable] Error: Recursion limit reached while instantiating 'simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.Variable_Ground_Temperature.simCenter.vleFluid_amb1.vleFluidType.fixedMixingRatio'.
`

**Comment by adrpo** [comment]
@perost I guess one can check that is the same instance node for `simCenter` and `simCenter.Variable_Ground_Temperature.simCenter`  which should trigger a lookup stop.

**Comment by adrpo** [comment]
I would basically ignore the `simcenter` inside `Variable_Ground_Temperature`.

**Comment by adrpo** [comment]
@AnHeuermann any idea if they have a version of TILMedia where they fixed the recursive inner/outer?

**Comment by perost** [comment]
Possibly fixed in #10919, by simply not instantiating outer components at all. I'm actually not sure why we did instantiate them earlier since we resolve references to them to the inner elements anyway, but we'll see if it breaks anything. As a bonus this also significantly speeds up instantiation of models that heavily use inner/outer (such as the BRSL ones).

The model I tested (`TransiEnt.Storage.Gas.Check.TestGasStorage_varXi`) now instead fails with `Error: Modified element value not found in class Real`, which I'm just going to classify as "won't fix" since it's expressly forbidden.
